projects/core/src/window/window-ref.ts

```diff
@@ -42,6 +45,30 @@ export class WindowRef {
     return this.isBrowser() ? window : undefined;
   }
 
+  /**
+   * Returns a reference to the document contained in the window or mimics the document when
+   * the process runs outside the browser.
+   *
+   * The later is useful, when the process runs in SSR and the document is lacking location properties
+   * such as href and origin. These are constructed by leveraging the _injected_ `SERVER_REQUEST_URL`
+   * and `SERVER_REQUEST_ORIGIN`.
+   *
+   * Please be aware that in SSR the document is still lacking the full fletched document object.
+   */
+  get document(): Document {
+    if (!this.isBrowser()) {
+      return {
+        ...this._document,
+        location: {
+          href: this.serverRequestUrl,
+          origin: this.serverRequestOrigin,
+        },
+      } as Document;
+    } else {
+      return this._document;
+    }
+  }
```

Comment:
[Major] It doesn't look safe to manually patch the `Document` properties in SSR anyway. Why:
- Since Angular Universal doesn't set all the `document` properties, maybe it has a reason not to do it?
- If Angular Universal will add the properties, our patched implementation would shadow it
- `location` is also available in `window`. Why we're not patching `window.location` too?

What about adding a separate getter property in `WindowRef`: `get location() { ... } ` which could return object with `href` and `origin`? We already have kinda' similar `get localStorage() { ... }`

Reply:
That's a nice suggestion, I've implemented accordingly.